### PR TITLE
Diverse build.info: Adjust paths

### LINKED ISSUE #12815: Empty directories
I've noted that when build OpenSSL using MSVC, some empty directories are created:
```
  ./crypto/include
  ./crypto/providers
```

I configured with:
```
  perl.exe -I. -I./util/perl Configure VC-WIN32-ONECORE no-makedepend \
  enable-zlib enable-trace enable-md2 no-ec_nistp_64_gcc_128 \
  enable-asan enable-aria enable-rc5 enable-heartbeats --prefix=/foo/bar --openssldir=/foo/bar
```

I see that the generated `Makefile` has lots of `-I "crypto\include"`. AFAICS needlessly.<br>
But I see no `mkdir crypto\include` anywhere. And I don't see `mkdir-p.pl` is used to create these. 

I mailed about this issue [here](https://mta.openssl.org/pipermail/openssl-users/2020-September/012956.html) also.  Quoting from Dr Mathias [reply](https://mta.openssl.org/pipermail/openssl-users/2020-September/012957.html):

>  it seems like the empty directories are created by the Configure script. 
> ...

Caused by these https://github.com/openssl/openssl/commit/25f2138b0ab54a65ba713c093ca3734d88f7cb51 and https://github.com/openssl/openssl/pull/9333

### PR DESCRIPTION
**Pull request by levitte**
Fixes #12815

### CODE REVIEW

File: crypto/poly1305/build.info [hunk 1]
```diff
@@ -34,7 +34,7 @@ SOURCE[../../libcrypto]=poly1305.c $POLY1305ASM
 # Implementations are now spread across several libraries, so the defines
 # need to be applied to all affected libraries and modules.
 DEFINE[../../libcrypto]=$POLY1305DEF
-DEFINE[../providers/libimplementations.a]=$POLY1305DEF
+DEFINE[../../providers/libimplementations.a]=$POLY1305DEF
```
Comment (paulidale) [inline crypto/poly1305/build.info:37]: This doesn't look related?
It's correct though.

**Review by paulidale — APPROVED** [review]

File: crypto/poly1305/build.info (hunk 1 above)
Comment (levitte) [reply crypto/poly1305/build.info:37]: It is, it caused the creation of crypto/providers

Comment (paulidale) [reply crypto/poly1305/build.info:37]: Ah!  Makes sense.  It was definitely wrong the way it was :)

File: test/build.info [hunk 2]
```diff
@@ -685,7 +685,7 @@ IF[{- !$disabled{tests} -}]
     DEPEND[keymgmt_internal_test]=../libcrypto.a libtestutil.a
 
     SOURCE[ffc_internal_test]=ffc_internal_test.c
-    INCLUDE[ffc_internal_test]=.. ../include ../apps/include ../crypto/include
+    INCLUDE[ffc_internal_test]=.. ../include ../apps/include ../include/crypto
```
Comment (mspncp) [inline test/build.info:688]: Without having tested it, I'm quite sure that you can omit the `../include/crypto` directory entirely, for two reasons:

Short reason:  It compiled without, so why add it?

Detailed reason: one of the goals of the refactoring was to reduce the amount of private include directory paths to only one, namely `../include` and have relative includes like the following instead. 

```C
#include "crypto/evp.h"
```
That's why `../include/crypto` is not necessary.

This approach had also the positive side effect to disambiguate between _private_ include directives (including header files from ../include, shared by several libcrypto subsystems) and _local_ include directives (including header files located in the same subdirectory as the C file).

So in fact, keeping the path `../include/crypto` is harmful, because it restores ambiguity: the same file could now be included as 

```C
#include "evp.h"
```

**Review by mspncp — CHANGES_REQUESTED** [review]
Except for one change, it looks good.

File: test/build.info (hunk 2 above)
Comment (paulidale) [reply test/build.info:688]: +1

Comment (levitte) [reply test/build.info:688]: Fixed

**Review by mspncp — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by levitte** [comment]
Merged

b830e0042972a237c6677c071f1fcde5c1afbea7 Diverse build.info: Adjust paths
